### ISSUE TITLE
Installing euclid on older R versions

### ISSUE THREAD

**Issue by venkatachalapathy** [thomasp85/euclid#25]
So I'm trying to install `euclid` on my windows machine that has `R.3.6.3` installed. `devtools` is installed and working fine.  My suspicion is `cgal` incompatibility with `mingw` , but can this be sorted purely at the `R` level? 

> C:/Users/.../R/R-3.6.3/library/cgal4h/include/CGAL/internal/enable_third_party_libraries.h:39:6: warning: #warning "<gmp.h> cannot be found. Less efficient number types will be used instead. Define CGAL_NO_GMP=1 if that is on purpose." [-Wcpp]
>  \#    warning "<gmp.h> cannot be found. Less efficient number types will be used instead. Define CGAL_NO_GMP=1 if that is on purpose."
>       ^
> In file included from C:/Users/.../R/R-3.6.3/library/cgal4h/include/CGAL/Rational_traits.h:32:0,
>                  from C:/Users/.../R/R-3.6.3/library/cgal4h/include/CGAL/number_type_basic.h:49,
>                  from C:/Users/.../R/R-3.6.3/library/cgal4h/include/CGAL/basic.h:45,
>                  from C:/Users/.../R/R-3.6.3/library/cgal4h/include/CGAL/Cartesian/Cartesian_base.h:29,
>                  from C:/Users/.../R/R-3.6.3/library/cgal4h/include/CGAL/Simple_cartesian.h:29,
>                  from C:/Users/.../R/R-3.6.3/library/cgal4h/include/CGAL/Exact_predicates_exact_constructions_kernel.h:29,
>                  from ../inst/include/internal/cgal_types.h:5,
>                  from api.cpp:7:
> C:/Users/.../R/R-3.6.3/library/cgal4h/include/CGAL/is_convertible.h:29:19: fatal error: gmpxx.h: No such file or directory
>  #include <gmpxx.h>
>                    ^
> compilation terminated.
> make: *** [C:/Users/.../DOCUME~1/R/R-36~1.3/etc/i386/Makeconf:215: api.o] Error 1
> ERROR: compilation failed for package 'euclid'

**Comment by dickoa** [comment]
Hi @venkatachalapathy 

At the R level it might be possible to change the configure script of `cgal4h` to check for `gmp` and if it's not present use the `CGAL_NO_GMP=1` flag. 
 
Did you try to install `gmp` using `mingw`

```
mingw-get install mingw32-gmp mingw32-mpfr
```

Thanks

**Comment by venkatachalapathy** [comment]
I used 

> remotes::install_github("thomasp85/euclid")

Peeking into the aborted install directory
> C:\Users\...\Documents\R\R-3.6.3\library\cgal4h

I don't find any configuration sript files. 

Also, I don't have `mingw` installed explicitly in my system. Do you want me to install `mingw` or use `mingw` installed when `RTools`  was installed?

**Comment by venkatachalapathy** [comment]
Thanks. I'm using the older version just to learn to navigate through such incompatibility issues. 

I see that the latest version of `Rtools` has a `mingw bash`. My older `Rtools` does not have a bash, and requires using R command prompt to execute, I think. 

So your recommendation is to `git clone` and not use `remotes::install_github()` option? In my case, I would have to 

+ download a `.tar.gz` file or `git clone`
+ use `install.packages()` with source flag turned on and make sure that I add the `CGAL_NO_GMP=1` in the `configure` file

Does this sound right to you?

**Comment by dickoa** [comment]
You can clone or download the code source and just change the `Makevars.win` file (no need to tweak `configure` here).  Try to add the proper flag there (i.e `CGAL_NO_GMP=1`) and make sure to remove the link to gmp/mpfr library. 

https://github.com/thomasp85/euclid/blob/master/src/Makevars.win

Once edited your can use `devtools::install` to install from your local source.
Let me know if it works.

Ahmadou

**Comment by venkatachalapathy** [comment]
No luck. 
 
 So I downloaded the repo to `"C:/Users/.../Documents/local_R_packages/euclid-master` and deleted line 5 of `Makevars.win`, and included the flag as you suggested. 

It looks like 

> CXX_STD = CXX11
> 
> PKG_CPPFLAGS=-DCGAL_HEADER_ONLY -DCGAL_USE_GMPXX -DBOOST_NO_AUTO_PTR -I../inst/include/internal/
> 
> CGAL_NO_GMP=1
> 

 
 I ran the following code in R shell in the above directory
 
 > install.package(".") 

and all other combinations like `devtools::install_local()` or `install_local()` etc. All give the same error. Does the following error make sense?

 > installation of package 'C:/Users/.../AppData/Local/Temp/RtmpqU7h8Q/fileb7685a7421c9/euclid_0.0.0.9000.tar.gz' had non-zero exit status
 
 If the package has been downloaded locally, why would R refer to `../Temp/..` directory?
